### ISSUE TITLE
Incorrect "undefined variable" error for named tuple class variables as return type annotations

### ISSUE THREAD

**Issue by aborealis** [PyCQA/pylint#5568]
### Bug description

pylint seems to incorrectly identify an undefined variable error when using a namedtuple class variable as a return type annotation. Argument type annotations appear to be OK.

### Steps to reproduce
```
# main.py
from collections import namedtuple

class MyObject:
    Coords = namedtuple('Point', ['x', 'y'])
 
    def my_method(self) -> Coords:
        pass
```

### Command used

```shell
pylint
```

### Pylint output

```shell
************* Module main
main.py:161:43: E0602: Undefined variable 'Coords' (undefined-variable)
```

### Expected behavior

no errors

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.9.9 (main, Nov 16 2021, 10:24:31) 
[GCC 11.2.0]
```

### OS / Environment

Kali Liniux
